### PR TITLE
[SPARK-34539] [BUILD] [INFRA] Remove stand-alone version Zinc server.

### PR DESCRIPTION
**Pull request by Yikun**
### What changes were proposed in this pull request?
Cleanup all Zinc standalone server code, and realated coniguration.

### Why are the changes needed?
image
- Zinc is the incremental compiler to speed up builds of compilation.
- The scala-maven-plugin is the mave plugin, which is used by Spark, one of the function is to integrate the Zinc to enable the incremental compiler.
- Since Spark v3.0.0 ([SPARK-28759](https://issues.apache.org/jira/browse/SPARK-28759)), the scala-maven-plugin is upgraded to v4.X, that means Zinc v0.3.13 standalone server is useless anymore. 

However, we still download, install, start the standalone Zinc server. we should remove all zinc standalone server code, and all related configuration.

See more in [SPARK-34539](https://issues.apache.org/jira/projects/SPARK/issues/SPARK-34539) or the doc [Zinc standalone server is useless after scala-maven-plugin 4.x](https://docs.google.com/document/d/1u4kCHDx7KjVlHGerfmbcKSB0cZo6AD4cBdHSse-SBsM).

### Does this PR introduce _any_ user-facing change?
No
    

### How was this patch tested?
Run any mvn build:
./build/mvn -DskipTests clean package -pl core
You could see the increamental compilation is still working, the stage of "scala-maven-plugin:4.3.0:compile (scala-compile-first)" with incremental compilation info, like:
```
[INFO] --- scala-maven-plugin:4.3.0:testCompile (scala-test-compile-first) @ spark-core_2.12 ---
[INFO] Using incremental compilation using Mixed compile order
[INFO] Compiler bridge file: /root/.sbt/1.0/zinc/org.scala-sbt/org.scala-sbt-compiler-bridge_2.12-1.3.1-bin_2.12.10__52.0-1.3.1_20191012T045515.jar
[INFO] compiler plugin: BasicArtifact(com.github.ghik,silencer-plugin_2.12.10,1.6.0,null)
[INFO] Compiling 303 Scala sources and 27 Java sources to /root/spark/core/target/scala-2.12/test-classes ...
```

### CODE REVIEW

**Comment by srowen** [comment]
Jenkins test this please

File: pom.xml [hunk 1]
```diff
@@ -2534,7 +2534,6 @@
             <checkMultipleScalaVersions>true</checkMultipleScalaVersions>
             <failOnMultipleScalaVersions>true</failOnMultipleScalaVersions>
             <recompileMode>incremental</recompileMode>
-            <useZincServer>true</useZincServer>
```
Comment (srowen) [inline pom.xml:2537]: Is this a default now? I guess we'll look at the build logs to ensure zinc is actually used

Comment (Yikun) [reply pom.xml:2537]: The [`useZincServer` has been removed](https://github.com/davidB/scala-maven-plugin/pull/321/files#diff-7d0815d86a5ea5598e1ee37c18a5592e90820c22ea61202bd83ae0dba066225dL83-L84) and the [incremental feature is enable by default](https://github.com/davidB/scala-maven-plugin/commit/0f01f2fe37aaadbefb02fb713d469dbd10acf8b1), in scala-maven-plugin 4.X. and the arg [recompileMode](https://github.com/davidB/scala-maven-plugin/blob/d7d19a2e98e7661d30fadb726fb0e91177eee5b6/src/main/java/scala_maven/ScalaCompilerSupport.java#L45-L50) is used to enable or disable the increamtal compiler.

**Comment by srowen** [comment]
Jenkins retest this please

**Comment by srowen** [comment]
Note to self, if we merge this, we need to update the docs about zinc on spark-website too.

**Comment by HyukjinKwon** [comment]
Yeah, I noticed there's something fishy about this (SPARK-28759). Thanks for taking a look and fixing this. cc @tgravescs and @dongjoon-hyun as well from that JIRA.

File: .github/workflows/build_and_test.yml [hunk 2]
```diff
@@ -99,12 +99,11 @@ jobs:
       if: ${{ github.event.inputs.target != '' }}
       run: git merge --progress --ff-only origin/${{ github.event.inputs.target }}
     # Cache local repositories. Note that GitHub Actions cache has a 2G limit.
-    - name: Cache Scala, SBT, Maven and Zinc
+    - name: Cache Scala, SBT, Maven
```
Comment (HyukjinKwon) [inline .github/workflows/build_and_test.yml:102]: nit .. but `Cache Scala, SBT and Maven`

**Comment by Yikun** [comment]
> Note to self, if we merge this, we need to update the docs about zinc on spark-website too.

@srowen I updated the doc:
- common part: https://github.com/apache/spark-website/pull/312, it can be merged directly, because spark is using the scala-maven-plugin 4.X with zinc 1.X
- security part: https://github.com/apache/spark-website/pull/313, it can be merged after this PR, and I'm not sure any other special process when update the security doc, so I put it into a separated patch.

File: .github/workflows/build_and_test.yml (hunk 2 above)
Comment (Yikun) [reply .github/workflows/build_and_test.yml:102]: oops, will be addressed in next PS.

**Comment by srowen** [comment]
Jenkins retest this please

**Comment by srowen** [comment]
Any further comments? I'd like to merge it.

**Comment by srowen** [comment]
Merged to master

**Comment by HyukjinKwon** [comment]
Looks good to me. Thanks for fixing this @Yikun

**Pull request closed without merging**